packages/fxa-settings/src/lib/sensitive-data-client.ts

```diff
@@ -6,31 +6,87 @@
 
 import { V1Credentials, V2Credentials } from './gql-key-stretch-upgrade';
 
-export const AUTH_DATA_KEY = 'auth';
+export namespace SensitiveData {
+  /**
+   * The keys for the various kinds of data that can be inserted into a `SensitiveData` object.
+   *
+   * @see SensitiveDataClient for implementations.
+   */
+  export enum Key {
+    Auth = 'auth',
+    AccountReset = 'accountResetData',
+    NewRecoveryKey = 'newRecoveryKeyData',
+    Password = 'password',
+    KeyStretchUpgrade = 'keyStretchUpgrade'
+  }
+
+  /**
+   * A data type for the various data added through the {@link SensitiveDataClient}.
+   *
+   * N.B. Use a non-nullable type if the values are derived from a hook during initialization.
+   *
+   * @see SensitiveDataClient for implementations.
+   */
+  export type DataMap = {
+    auth?: AuthData;
+    accountResetData?: AccountResetData;
+    newRecoveryKeyData?: NewRecoveryKeyData;
+    password?: Password;
+    keyStretchUpgrade?: KeyStretchUpgradeData;
+  };
 
-export type SensitiveDataClientData = {
-  [AUTH_DATA_KEY]: {
+  /**
+   * Data inserted for the key {@link Key.Auth}.
+   */
+  export type AuthData = {
     emailForAuth?: string;
     authPW?: string;
     keyFetchToken?: hexstring;
     unwrapBKey?: hexstring;
   };
-};
 
-export type SensitiveDataClientAuthKeys = Pick<
-  SensitiveDataClientData[typeof AUTH_DATA_KEY],
-  'keyFetchToken' | 'unwrapBKey'
->;
+  /**
+   * Data insert for the key {@link Key.Password}.
+   */
+  export type Password = {
+    plainTextPassword: string;
+  };
+
+  /**
+   * Data inserted for the key {@link Key.AccountReset}.
+   */
+  export type AccountResetData = {
+    keyFetchToken: string;
+    unwrapBKey: hexstring;
+  };
+
+  /**
+   * Data inserted for the key {@link Key.NewRecoveryKey}.
+   */
+  export type NewRecoveryKeyData = {
+    recoveryKey: Uint8Array;
+  };
+
+  /**
+   * Data inserted for the key {@link Key.KeyStretchUpgrade}.
+   */
+  export type KeyStretchUpgradeData = {
+    email: string;
+    v1Credentials: V1Credentials;
+    v2Credentials: V2Credentials;
+  };
+}
 
 /**
- * Class representing a client for handling sensitive data.
+ * A client that holds data which should never be persisted.
+ *
+ * N.B. This data only lives in memory and will always be lost after the document is unloaded. Avoid nullable data where possible.
  *
- * @class
+ * @class SensitiveDataClient
  */
 export class SensitiveDataClient {
   /**
-   * Object to store sensitive data.
-   * @private
+   * @deprecated
```

Comment:
Are these still remaining uses of `sensitiveData` that will be handled in a follow up, or can these now be removed?

Reply:
Yup, FXA-10849!